### DISCUSSION TITLE
Maintain `--dev / -D` for dev dependencies

### DISCUSSION THREAD

**Discussion by ccrvlh** [python-poetry/poetry#7530] [Ideas]
As I understand, the old way of declaring dependencies is being deprecated, favoring a more flexible approach where one could have many `group` of dependencies, so the new recommended approach becomes `-G dev`.

As much as one could have multiple varying groups of dependencies (`test`, `typed`, `dev`, `local` etc etc), probably one of the most common operations would be whether to add a project dependency (`poetry add pandas`) or just a development dependency `poetry add mypy`).

`--dev / -D` is arguably a very intuitive way of doing this and with less friction (when compared to `dev` being a `group` of dependencies), some package managers from other languages (`yarn` uses `--dev / -D` while `npm` uses `--save-dev`) and `pipenv` also has a `--dev` option. 

As much as I understand the option for a more flexible `--group` approach, I would like to suggest keeping the `--dev / -D` as shortcut - it's a great convenience flag, a lot easier/intuitive to use, and don't really see many downsides to it.

**Comment by wxgeo** [comment]
I totally agree with previous comment.

While I use _very_ often `poetry add -D <new_lib>`, I never use any other group than dev, and I expect many users to be in the same situation.

From a user point of view, much of poetry strength lies in its intuitive interface, and using `-D` or `--dev` seems both more efficient and more natural to me.

I can't see any disadvantage to keeping `-D/--dev` as a shortcut to `--group dev`.

**Comment by Askir** [comment]
I like the idea of groups but I have to admit that 90%+ of python projects that I've worked on only ever will have dev as a group so I would never have to worry about the group functionality. So +1 for keeping the convenience flags.
